papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/controller/ProjectsController.java

```diff
@@ -90,12 +95,75 @@ private Project getValidProject(Long id) {
         List<SprintState> sprintStates = sprintStateMapper.mapFromApi(
                 Optional.ofNullable(sprintStateDtos).orElse(ALL_SPRINT_STATE_DTOS));
 
-        List<SprintDto> sprints = sprintService.findByStatesInProject(sprintStates, project.getId(), currentTime)
+        List<SprintDto> sprints = sprintService.getByStatesInProject(sprintStates, project.getId(), currentTime)
                 .stream()
                 .sorted(Comparator.comparing(sprint -> sprint.getEnrollmentPeriod().getStart()))
                 .map(sprint -> sprintMapper.mapToApi(sprint, currentTime))
                 .collect(Collectors.toList());
 
         return ResponseEntity.ok(sprints);
     }
+
+    @Override
+    public ResponseEntity<AvailabilityDto> getUserAvailability(Long projectId, Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+
+        if (!isSprintInProject(sprint, project)) {
+            throw new ForbiddenAccessException();
+        }
+
+        Availability availability = availabilityService
+                .getBySprintIdAndUserId(sprintId, userContext.getUserIdn())
+                .orElseThrow(ResourceNotFoundException::new);
+
+        return ResponseEntity.ok(availabilityMapper.mapToApi(availability));
+    }
+
+    @Override
+    public ResponseEntity<AvailabilityDto> updateUserAvailability(@Valid AvailabilityDto body, Long projectId,
+            Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+        if (!isSprintInProject(sprint, project) || !isAvailabilityDeclarationAllowed(sprint)
+                || !body.getUserId().equals(userContext.getUserIdn())) {
```

Comment:
Jak to `userId` jednak nie jest required, to w sumie nie wiem jak lepiej - wymagać przy PUT poprawnego, czy akceptować puste - ewentualnie można dopuszczać oba scenariusze, ale nie wiem

Reply:
możemy założyć, że jest tylko do komunikacji backend -> frontend, więc tutaj byłoby ignorowane